adapters/tlsio_openssl.c

```diff
@@ -850,8 +852,12 @@ static int load_cert_crl_http(
         goto error;
     }
 
-    bio = BIO_new_connect(host);
-    if (!bio || !BIO_set_conn_port(bio, port))
+    const char* proxyHostnamePort;
+    const char* usernamePassword;
+    platform_get_http_proxy(&proxyHostnamePort, &usernamePassword);
+
+    bio = BIO_new_connect((proxyHostnamePort && *proxyHostnamePort) ? proxyHostnamePort : host);
+    if (!bio || ((!proxyHostnamePort || !(*proxyHostnamePort)) &&!BIO_set_conn_port(bio, port)))
```

Comment:
please move this to a separate check. you are already doing abosolutely the same check 2 lines up:
bool useProxyPort = proxyHostnamePort && * proxyHostnamePort;
bio = ... (useProxyPort ? .. : host)
...
if(!bio || (!useProxyPort && ...))